gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/AssetType.java

```diff
@@ -640,15 +646,17 @@ public boolean unlock() {
         public String toString() {
             return "lock(" + urn + ")";
         }
+
     }
 
-    private static final class AssetReference<T> extends PhantomReference<T> {
+    private final class AssetReference<T extends Asset<?>> extends PhantomReference<T> {
```

Comment:
Why is this not static? The implicit reference to the outer object doesn't seem to be used.